### PR TITLE
syncer: refine heartbeat

### PR DESCRIPTION
**Pull request by csuzhangxc**
### What problem does this PR solve? 

- #706
- #694 
- #707

### What is changed and how it works?

- compare heartbeat schema and table as case-insensitive
- do not update replication lag metric if no valid slave ts exit
- use `UTC_TIMESTAMP(6)` to replace `UTC_TIMESTAMP()`

### Check List 

Tests 

 - Unit test
 - Integration test

Related changes

 - Need to be included in the release note

### CODE REVIEW

**Comment by csuzhangxc** [comment]
@GMHDBJD PTAL

File: syncer/heartbeat.go [hunk 1]
```diff
@@ -292,8 +293,7 @@ func (h *Heartbeat) createTable() error {
 
 // updateTS use `REPLACE` statement to insert or update ts
 func (h *Heartbeat) updateTS() error {
-	// when we not need to support MySQL <=5.5, we can replace with `UTC_TIMESTAMP(6)`
```
Comment (WangXiangUSTC) [inline syncer/heartbeat.go:295]: Does this mean we don't support MySQL <=5.5 after this pr?

File: tests/all_mode/run.sh [hunk 2]
```diff
@@ -101,8 +101,8 @@ function run() {
     # use sync_diff_inspector to check data now!
     check_sync_diff $WORK_DIR $cur/conf/diff_config.toml
 
-    check_metric $WORKER1_PORT 'dm_syncer_replication_lag{task="test"}' 3 0 1
-    check_metric $WORKER2_PORT 'dm_syncer_replication_lag{task="test"}' 3 0 1
+    check_metric $WORKER1_PORT 'dm_syncer_replication_lag{task="test"}' 3 0 2
```
Comment (WangXiangUSTC) [inline tests/all_mode/run.sh:104]: why need to increase the `upper` to `2`? It seems that use `UTC_TIMESTAMP(6)` to replace `UTC_TIMESTAMP` when update ts will decrease the replication lag.

File: syncer/heartbeat.go (hunk 1 above)
Comment (csuzhangxc) [reply syncer/heartbeat.go:295]: YES, but we already highlighted this in docs (`5.5 < MySQL 版本 < 8.0`).

File: tests/all_mode/run.sh (hunk 2 above)
Comment (csuzhangxc) [reply tests/all_mode/run.sh:104]: I saw lag with 1.x (but < 1.5) in some cases. in the old `UTC_TIMESTAMP` it `==1`, but now it `> 1`.

**Comment by WangXiangUSTC** [comment]
LGTM

**Pull request merged**